languages/ruby/exercises/concept/basics/lasagna.rb

```diff
@@ -1,13 +1,13 @@
 class Lasagna
   def remaining_minutes_in_oven(actual_minutes_in_oven)
-    raise NotImplementedError, 'Please implement the remaining_minutes_in_oven method'
+    raise NotImplementedError, 'Please implement the Lasagna#remaining_minutes_in_oven method'
```

Comment:
I don't think this is a fair change.  There is no requirement that the `Lasagna#remaining_minutes_in_oven` method be implemented, only that the method needs to come into that scope.

Is that important for this exercise?  Is it something that we are teaching for this concept exercise, specifically?  I will have to look.

The test itself will verify that the method is callable at that scope.  So I am not sure the class needs to be stated here.

Reply:
The of this specifically being in the context of the class leaves little doubt that the method is called on the instance as well.  Indeed, for maintenance sake, we should use `#{self.class}#remaining_minutes_in_oven` so the class name, if ever changed, will remain correct.

Reply:
Fair point - looking through the [instructions.md](https://github.com/exercism/v3/blob/master/languages/ruby/exercises/concept/basics/.docs/instructions.md) it looks like we specify methods like: `Lasagna#remaining_minutes_in_oven`. This is one of the reasons I made this change.

Agreed that `#{self.class}#remaining_minutes_in_oven` makes it more maintainable. But if we change the classname wouldn't we have to manually change those references in the docs? Happy to make this change though! What are your thoughts?

Reply:
Yes, but we can do things like introspection or refer to the generated tests to verify.  So the maintenance is not gone, just reduced.

Also, if you look down through this source, the one that does make sense to specifically state `#initialize` is, well, `initialize`.  This won't come from a module, for example.

Reply:
For the context of where this exercise is, the student wouldn't understand what `self.class` is so I don't think we can use it in a public-facing file here.  

One of the generally challenges with creating these Concept Exercises is limiting ourselves to what a student has been already taught and not introducing other syntax/methods that might overwhelm them.

Reply:
That is very true, very much the case.  No eggs and chickens here.   And this is not a generator thing either, which means that we are kind of stuck with the maintenance if the class were to change as well (since we also aren't dynamically generating this file).

I was thinking more of "not reading the source" but getting that message in the runtime when it is raised.  So the time of reading is important, because once it is rendered, the translation becomes what is literally here.  Before it is rendered, the literal is what will be rendered.